modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/sync/SyncManagerImpl.kt

```diff
@@ -546,10 +547,8 @@ class SyncManagerImpl @Inject constructor(
 
     private fun refreshToken(): Single<AccessToken> {
         syncAccountManager.invalidateAccessToken()
-        return Single.fromCallable {
-            runBlocking {
-                syncAccountManager.getAccessToken()
-            } ?: throw RuntimeException("Failed to get token")
+        return rxSingle {
+            syncAccountManager.getAccessToken() ?: throw RuntimeException("Failed to get token")
```

Comment:
This isn't related to the issue but removing runBlocking seems to be a good idea.